components/webcompat_reporter/browser/webcompat_report_uploader.cc

```diff
@@ -25,29 +26,46 @@
 #include "url/gurl.h"
 #include "url/origin.h"
 
-namespace brave {
+namespace webcompat_reporter {
 
 WebcompatReportUploader::WebcompatReportUploader(
     scoped_refptr<network::SharedURLLoaderFactory> factory)
     : shared_url_loader_factory_(std::move(factory)) {}
 
 WebcompatReportUploader::~WebcompatReportUploader() = default;
 
-void WebcompatReportUploader::SubmitReport(const GURL& report_url,
-                                           const base::Value& details,
-                                           const base::Value& contact) {
+void WebcompatReportUploader::SubmitReport(
+    const GURL& report_url,
+    const bool shields_enabled,
+    const std::string& ad_block_setting,
+    const std::string& fp_block_setting,
+    const std::string& ad_block_list_names,
+    const std::string& languages,
+    const bool language_farbling,
+    const bool brave_vpn_connected,
+    const base::Value& details,
+    const base::Value& contact) {
   std::string api_key = brave_stats::GetAPIKey();
 
   const GURL upload_url(BUILDFLAG(WEBCOMPAT_REPORT_ENDPOINT));
 
   url::Origin report_url_origin = url::Origin::Create(report_url);
 
   base::Value::Dict post_data_obj;
-  post_data_obj.Set("url", base::Value(report_url.spec()));
-  post_data_obj.Set("domain", base::Value(report_url_origin.Serialize()));
-  post_data_obj.Set("additionalDetails", details.Clone());
-  post_data_obj.Set("contactInfo", contact.Clone());
-  post_data_obj.Set("api_key", base::Value(api_key));
+  post_data_obj.Set(kSiteURLField, report_url.spec());
```

Comment:
So we are sending the full url?

Reply:
we remove the query and fragment here:

https://github.com/brave/brave-core/pull/20052/files#diff-a2c14ce0b115f48dd789f026367f25e06f48d83178025b78c43c98e3126ae72dR128-R133

Reply:
I see.  Would you mind adding a comment here saying so then, since both the field name and the method being called make it seem (to the naive reader) that the full site URL is being called.

Or rename `report_url` to make the same clear (e.g., `report_origin_and_path` or similar)